storage/__init__.py

```diff
@@ -157,9 +169,12 @@ def _sign(self, params, url, meth):
         return b64encode(hmac.new(key, sigstr, hashmod).digest())
 
     def _auth(self, meth, url):
-        params = {"ts": str(int(time.time()) + self.timeskew)}
+        ts = time.time()
+        if ts >= self.auth_expires_at:
+            self.regenerate()
```

Comment:
This puts a network call down here in the helper function, do we have to add some `async` or `await` or something here?  Sorry, I'm not all up to speed on the async-IO stuff.

Reply:
No worries, this one is done synchronously - we might see a freeze in the RPS when it happens, but we have to stop all other calls anyways to wait for a new token. However we might need to add a flag so if another coroutine that's in the retry loop calls regenerate() again, gets the info that it's being taken care of.